src/main/java/it/uniroma3/mat/extendedset/intset/ImmutableConciseSet.java

```diff
@@ -352,9 +399,12 @@ public int compare(WordHolder h1, WordHolder h2)
 
           i.advanceTo(itr.wordsWalked);
           if (i.hasNext()) {
-            wordsToAdd.add(new WordHolder(i.next(), i));
+            i.word = i.next();
+            qIndex++;
+          } else {
+            removeElement(theQ, qIndex, qSize);
```

Comment:
> For additional optimization

I meant could we take _theQ[start]_ (instead of _theQ[0]_) and update _start_ instead of array copying?